grails-app/services/org/pih/warehouse/inventory/StockMovementService.groovy

```diff
@@ -675,16 +675,14 @@ class StockMovementService {
                 'max': max ? max.toInteger() : null,
         ]);
 
-        def editItemsIds = data.collect { it.id }
+        def editItemsIds = data.collect { "'$it.id'" }.join(',')
 
         def substitutionItemsMap = dataService.executeQuery("""
                     select
                        *
                     FROM substitution_item
-                    where parent_requisition_item_id in (:ids)
-                    """, [
-                'ids': editItemsIds,
-        ]).inject([:]) {map, item -> map << [(item.parent_requisition_item_id): item]}
+                    where parent_requisition_item_id in (${editItemsIds})
```

Comment:
I still don't really understand why we can't use the more common syntax that was already there. I don't mind you doing this
```
def editItemsIds = data.collect { "'$it.id'" }
```
because I understand you're trying to wrap each ID with quotes, but why can't you then pass that into the query as a parameter like we were already doing?

```
def substitutionItemsMap = dataService.executeQuery("""
select *
FROM substitution_item
where parent_requisition_item_id in (:ids)""", ["ids":editItemIds])
```

Reply:
Can you also explain the reason we need to replace inject by groupBy?

Reply:
@jmiranda It didn't work while passing it into the query, that is really strange but it returned 0 rows. We need to replase inject by groupBy because there can be more than one substitution item and with inject it always returned only 1.